libs/ui/input/helm/src/lib/hlm-input.directive.ts

```diff
@@ -49,4 +58,36 @@ export class HlmInputDirective {
 	protected _computedClass = computed(() =>
 		hlm(inputVariants({ size: this._size(), error: this._error() }), this.userClass()),
 	);
+
+	ngControl: NgControl | null = this.injector.get(NgControl, null);
+
+	errorStateTracker: ErrorStateTracker;
+
+	private defaultErrorStateMatcher = inject(ErrorStateMatcher);
+	private parentForm = inject(NgForm, { optional: true });
+	private parentFormGroup = inject(FormGroupDirective, { optional: true });
+
+	errorState = computed(() => this.errorStateTracker.errorState());
+
+	constructor(@Inject(Injector) private injector: Injector) {
```

Comment:
Let's not mix the inject function and constructor injector

Reply:
Done!